### ISSUE TITLE
import/extensions ignores extension overrides if ignorePackages is set in object options

### ISSUE THREAD

**Issue by le0nik** [benmosher/eslint-plugin-import#1248]
node: v10.14.1
eslint: v5.10.0
eslint-plugin-import: v2.14.0

.eslintrc.js

```js
module.exports = {
  plugins: ['import'],
  rules: {
    'import/extensions': ['error', 'always', {
      ignorePackages: true,
      js: 'never',
    }]
  }
}
```

./src/index.js

```js
import app from './app';
```

Running eslint results in an error:

```
/Volumes/Projects/project/src/index.js
1:22  error  Missing file extension "js" for "./app"  import/extensions
```

The problem seems to be here https://github.com/benmosher/eslint-plugin-import/blob/1cd82eb27df85768fbd076e4ff6b7f36d6f652ce/src/rules/extensions.js#L36-L45

Plugin expects extensions overrides to be specified inside `pattern` property inside the options object if `ignorePackages` property is also specified. It's not documented [here](https://github.com/benmosher/eslint-plugin-import/blob/1cd82eb27df85768fbd076e4ff6b7f36d6f652ce/docs/rules/extensions.md) though.

So this configuration works correctly:

```js
{
  rules: {
    'import/extensions': ['error', 'always', {
      ignorePackages: true,
      pattern: { // wrap extensions in the `pattern` object
        js: 'never',
      }
    }]
  }
}
```

Either documentation is lacking or `pattern` property should not be expected in options.

**Comment by ljharb** [comment]
Ths doesn’t seem resolved yet?

**Comment by ivo-stefchev** [comment]
Oh, how I love explicit configs.
So the reason why this `module.exports = {
  plugins: ['import'],
  rules: {
    'import/extensions': ['error', 'always', {
      ignorePackages: true,
      js: 'never',
    }]
  }
}`
is because in order to apply the **js** part - you need to have **ingnorePackages** to be **undefined**. So basically if **ignorePackages** is present in the object - it disables all extension specific options. For me at least this makes very little sense.

**Comment by ivo-stefchev** [comment]
Also this defaultConfig is enum of three possible values.
```
"import/extensions": [<severity>, "never" | "always" | "ignorePackages"]
```
**ignorePackages** doesn't do anything when you set it, hence I propose to be removed from the enumValues and left only as boolean property part of the object.

**Comment by ljharb** [comment]
Seems like a good PR.
